tools/cli/NearObjectCli.cxx

```diff
@@ -237,63 +274,221 @@ NearObjectCli::AddSubcommandUwbMonitor(CLI::App* parent)
+    std::string dstMacAddressDescription = "2-byte short MAC address(es) of other device(s). If device is Controller, list NumberOfControlees mac addresses. If device is Controlee, list Controller mac address";
+    if (appConfigParamsData.macAddressMode == uwb::UwbMacAddressType::Extended) { // TODO: Enable macAddressMode to be set before checking this value. Otherwise, the default empty value will always be used
+        deviceMacAddressDescription = "8-byte extended MAC address of own device: e.g. 12:34:56:78:87:65:43:21";
+        dstMacAddressDescription = "8-byte extended MAC address(es) of other device(s). If device is Controller, list NumberOfControlees mac addresses. If device is Controlee, list Controller mac address";
+    }
+    rangeStartApp->add_option("--DeviceMacAddress", m_cliData->deviceMacAddressString, deviceMacAddressDescription)->capture_default_str()->required();
+    rangeStartApp->add_option("--DestinationMacAddress", m_cliData->destinationMacAddressString, dstMacAddressDescription)->capture_default_str()->required();
+
+    detail::AddEnumOption(rangeStartApp, appConfigParamsData.deviceType, true);
+
+    rangeStartApp->parse_complete_callback([this, rangeStartApp] {
+        // TODO: Move validation logic into its own function
+
+        // Validate NumberOfControlees
+        if (m_cliData->appConfigParamsData.multiNodeMode == uwb::protocol::fira::MultiNodeMode::Unicast) {
+            if (m_cliData->appConfigParamsData.numberOfControlees != 1) {
+                std::cerr << "Only 1 controlee expected in Unicast mode" << std::endl;
+            }
+        } else {
+            if (m_cliData->appConfigParamsData.numberOfControlees < 1 || m_cliData->appConfigParamsData.numberOfControlees > 8) {
+                std::cerr << "Invalid number of controlees. Must be 1 <= N <= 8" << std::endl;
+            }
+        }
+        // Set MAC addresses
+        const auto macAddressType = m_cliData->appConfigParamsData.macAddressMode == uwb::UwbMacAddressType::Extended ? uwb::UwbMacAddressType::Extended : uwb::UwbMacAddressType::Short;
+
+        m_cliData->appConfigParamsData.deviceMacAddress = uwb::UwbMacAddress::FromString(m_cliData->deviceMacAddressString, macAddressType);
+        m_cliData->appConfigParamsData.destinationMacAddresses.emplace();
+        // TODO: Insert the correct Controlee mac address rather than simply inserting the entire destinationMacAddressString
+        for (auto i = 0; i < m_cliData->appConfigParamsData.numberOfControlees; i++) {
+            m_cliData->appConfigParamsData.destinationMacAddresses.value().insert(uwb::UwbMacAddress::FromString(m_cliData->destinationMacAddressString, macAddressType).value());
+        }
+
+        std::cout << "Selected parameters:" << std::endl;
+        // TODO: Display selected parameters
+        std::cout << "DeviceRole: " << static_cast<int>(m_cliData->appConfigParamsData.deviceRole.value()) << std::endl;
```

Comment:
For each of these, why not display the enum name instead of the numerical integer value? eg. `magic_enum::enum_name(m_cliData->appConfigParamsData.deviceRole.value())`

Reply:
Ah yeah I was thinking that that would be better but I forgot about magic_enum. Thanks!